assignment01-voronoi/worldgen-voronoi.c

```diff
@@ -0,0 +1,224 @@
+{
+    return sqrt( (x-x2) * (x-x2) + (y-y2) * (y-y2));
+}
+
+void voronoi_world_gen(char gridMatrix[GRID_HEIGHT][GRID_WIDTH], Seed storedSeeds[SEED_NUM])
+{   
+    // start at one because of border
+    for (int x =1; x < GRID_HEIGHT - 1 ; x++){
+        for ( int y =1; y < GRID_WIDTH - 1; y++){
+            int closestSeed = 0;
+            double closestDist = distance(x , y, storedSeeds[0].x, storedSeeds[0].y);
+            for(int i=1; i < SEED_NUM; i++){
+                double currDist = distance(x , y, storedSeeds[i].x, storedSeeds[i].y);
+                if(currDist < closestDist){
+                    closestSeed = i;
+                    closestDist = currDist;
+                }
+            }
+            // TODO: this is not desired way to do this but it works
+            // do something similar to printing the colors
+            if(storedSeeds[closestSeed].tile == '^') {
+                int treeChance = rand() % 100;
+                if (treeChance < 25) {
+                    gridMatrix[x][y] = treeTile.ascii;
+                }
+                // else if(25 < treeChance && treeChance < 50) {
+                //     gridMatrix[x][y] = tallGrass.ascii;
+                // }
+                else {
+                    gridMatrix[x][y] = clearing.ascii;
+                }
+            }
+            else {
+                gridMatrix[x][y] = storedSeeds[closestSeed].tile;
+            }
+        }
+     }
+}
+
+void generate_path(char grid[GRID_HEIGHT][GRID_WIDTH], int leftGateX, int leftGateY, int rightGateX, int rightGateY,
+                    int topGateX, int topGateY, int botGateX, int botGateY) 
+{   // x is y, y should be x
+    int x = leftGateX;
+    int y = leftGateY;
+    int j = botGateX;
+    int k = botGateY;
+
+    while (y <= topGateY) {
+        grid[x][y] = path.ascii;
+        y++;
+    } y--;
+
+    if( x <= rightGateX){
+        while (x <=rightGateX){
+            grid[x][y] = path.ascii;
+            x++;
+        } x--;
+
+    } else if (x >= rightGateX){
+        while (x >= rightGateX)
+        {
+            grid[x][y] = path.ascii;
+            x--;
+        } x++;
+    }
+
+    while (y <= rightGateY){
+        grid[x][y] = path.ascii;
+        y++;
+    }
+
+    // down here is bot to top path looping
+    while(j>= rightGateX){
+        grid[j][k] = path.ascii;
+        j--;
+    } j++;
+
+    if( k <= topGateY){
+        while(k <= topGateY){
+            grid[j][k] = path.ascii;
+            k++;
+        } k--;
+    } else if (k >= topGateY){
+        while (k >= topGateY)
+        {
+            grid[j][k] = path.ascii;
+            k--;
+        }k++;
+    }
+    while(j >= topGateX){
+        grid[j][k] = path.ascii;
+        j--;
+    }
+}
+
+
+int main(void)
+{
+    srand(time(NULL));    
+    Seed seeds[SEED_NUM];
+
+    gen_voronoi_seeds(seeds);
+    init_world();
+    voronoi_world_gen(gridMatrix, seeds);
+
+    // okay so my grid is like inversed so top is left, bot is right...etc :)
```

Comment:
it is common that 0,0 is top left and WIDTH-1, HEIGHT-1 is bottom right

Reply:
I swear i got confused or something, everytime i worked with what i thought was my left gate x and y, it was changing my top gate???